qt/scientific_interfaces/Inelastic/Analysis/IndirectFitDataView.cpp

```diff
@@ -88,8 +113,11 @@ void IndirectFitDataView::setHorizontalHeaders(const QStringList &headers) {
 
   auto header = m_uiForm->tbFitData->horizontalHeader();
   header->setSectionResizeMode(0, QHeaderView::Stretch);
-  m_uiForm->tbFitData->setItemDelegateForColumn(headers.size() - 1,
-                                                std::make_unique<ExcludeRegionDelegate>().release());
+
+  m_uiForm->tbFitData->setItemDelegateForColumn(startXColumn(), std::make_unique<NumericInputDelegate>().release());
+  m_uiForm->tbFitData->setItemDelegateForColumn(endXColumn(), std::make_unique<NumericInputDelegate>().release());
+  m_uiForm->tbFitData->setItemDelegateForColumn(excludeColumn(), std::make_unique<ExcludeRegionDelegate>().release());
```

Comment:
Further to the point above, I think it will be more efficient to create a raw pointer rather than a unique pointer, and then releasing the unique pointer

Reply:
Ah, I see what you mean. I think maybe we could refactor the `startXColumn`, `endXColumn` and `excludeColumn` functions. Maybe have a private function that does
```
auto labels = m_uiForm->tbFitData->horizontalHeaderLabels();
return labels.indexOf(columnName);
```
and then each of the `startXColumn`, `endXColumn` and `excludeColumn`  functions would call `return columnIndexOf("StartX")` where we change "StartX" for each of the different column names. Then we could call these functions as before, and remove the overridden functions in the FqFitView and ConvFitView. If that makes sense?

Reply:
Thanks for the suggestion. I agree all the startX,etc.. functions can be refactored into one single one. 
The QTableWidget does not have any function to retrieve the horizontal header labels. The way to do it is a bit involved (basically iterating over all column headers). 
Since the headers string list is passed upon construction, I have stored the list as a private data member of the class, and then added a public function to retrieve the index position. 
Let me know what you think.